dask_planner/src/expression.rs

```diff
@@ -213,36 +215,36 @@ impl PyExpr {
     /// RexConverter plugin instance should be invoked to handle
     /// the Rex conversion
     #[pyo3(name = "getExprType")]
-    pub fn get_expr_type(&self) -> String {
-        String::from(match &self.expr {
-            Expr::Alias(..) => "Alias",
-            Expr::Column(..) => "Column",
-            Expr::ScalarVariable(..) => panic!("ScalarVariable!!!"),
-            Expr::Literal(..) => "Literal",
-            Expr::BinaryExpr { .. } => "BinaryExpr",
-            Expr::Not(..) => panic!("Not!!!"),
-            Expr::IsNotNull(..) => panic!("IsNotNull!!!"),
-            Expr::Negative(..) => panic!("Negative!!!"),
-            Expr::GetIndexedField { .. } => panic!("GetIndexedField!!!"),
-            Expr::IsNull(..) => panic!("IsNull!!!"),
-            Expr::Between { .. } => "Between",
-            Expr::Case { .. } => panic!("Case!!!"),
-            Expr::Cast { .. } => "Cast",
-            Expr::TryCast { .. } => panic!("TryCast!!!"),
-            Expr::Sort { .. } => "Sort",
-            Expr::ScalarFunction { .. } => "ScalarFunction",
-            Expr::AggregateFunction { .. } => "AggregateFunction",
-            Expr::WindowFunction { .. } => panic!("WindowFunction!!!"),
-            Expr::AggregateUDF { .. } => panic!("AggregateUDF!!!"),
-            Expr::InList { .. } => "InList",
-            Expr::Wildcard => panic!("Wildcard!!!"),
-            Expr::InSubquery { .. } => "Subquery",
-            Expr::ScalarUDF { .. } => "ScalarUDF",
-            Expr::Exists { .. } => "Exists",
-            Expr::ScalarSubquery(..) => "ScalarSubquery",
-            Expr::QualifiedWildcard { .. } => "Wildcard",
-            Expr::GroupingSet(..) => "GroupingSet",
-        })
+    pub fn get_expr_type(&self) -> PyResult<String> {
+        match &self.expr {
```

Comment:
Style wise would it make more since to move back to the `String::from(...)` at the match pattern level so individual `Ok(...)` are not needed and then have a single arm for all the "Err" Expr types that were previously `panic(..)`. Like this ....

```
Expr::ScalarVariable(..) 
| Expr::Not(..)
| Expr::IsNotNull(..)
| Expr::Negative(..)
| Expr::GetIndexedField {..}
| Expr::IsNull(..)
| Expr::Case{..}
| Expr::TryCast{..}
| Expr::WindowFunction{..}
| Expr::AggregateUDF{..}
| Expr::Wildcard => Err(py_type_err("Encountered unsupported expression type: {:?}", &self.expr))

Reply:
If we opt to do this, is there any reason you prefer specifying all the variants explicitly rather than using a catch all?

Reply:
Technically of course either way works. With lower level languages like Rust though the style usually opts for making things more verbose so the behavior is more clear and anyone can look at the code and clearly see the Expr variants that trigger the condition. With the default "catch all" its not immediately clear without consulting the documentation. Also if say a new Expr variant is added I think you would want the build to fail and not just default on its behavior and manually listing each variant achieves that. Honestly I think its just opinion based and either is fine.

Reply:
That logic is sound to me! Happy to stick to enumerating all variants, would you be opposed to doing this for some of the other matches in the expressions implementation where we currently use catch alls?